Managers/UInputMan.cpp

```diff
@@ -1,2858 +1,1393 @@
+				} else if (g_InActivity) {
+					// The mouse cursor is visible and can move about the screen/window, but it should still be contained within the mouse player's part of the window
+					ForceMouseWithinPlayerScreen(mousePlayer);
+				}
+			}
 
+			// Mouse wheel update, translate motion into discrete ticks
+			if (std::abs(mouse_z) >= 1) {
+				// Save the direction
+				m_MouseWheelChange = mouse_z;
+				// Reset the position
+				position_mouse_z(0);
+			} else {
+				m_MouseWheelChange = 0;
+			}
 
-void UInputMan::ClearAccumulatedStates()
-{
-	for (int element = 0; element < INPUT_COUNT; element++)
-	{
-		m_aNetworkAccumulatedElementPressed[element] = false;
-		m_aNetworkAccumulatedElementReleased[element] = false;
+			// Enable the mouse cursor positioning again after having been disabled. Only do this when the mouse is within the drawing area so it
+			// won't cause the whole window to move if the user clicks the title bar and unintentionally drags it due to programmatic positioning.
+			int mX = mouse_x / g_FrameMan.ResolutionMultiplier();
+			int mY = mouse_y / g_FrameMan.ResolutionMultiplier();
+			if (m_DisableMouseMoving && m_PrepareToEnableMouseMoving && (mX >= 0 && mX < g_FrameMan.GetResX() && mY >= 0 && mY < g_FrameMan.GetResY())) {
+				m_DisableMouseMoving = m_PrepareToEnableMouseMoving = false;
+			}
+		}
 	}
-}
 
-bool UInputMan::AccumulatedElementPressed(int element)
-{
-	if (element < 0 || element >= INPUT_COUNT)
-		return false;
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
-	return m_aNetworkAccumulatedElementPressed[element];
-}
+	void UInputMan::UpdateJoystickInput() {
+		// Detect and store joystick button changed since last Update()
+		for (int joystick = 0; joystick < num_joysticks; ++joystick) {
+			for (int button = 0; button < joy[joystick].num_buttons; ++button) {
+				s_ChangedJoystickStates[joystick].button[button].b = joy[joystick].button[button].b != s_PrevJoystickStates[joystick].button[button].b;
+			}
 
+			// Determine deadzone settings
+			float deadZone = 0.0F;
+			int deadZoneType = DeadZoneType::CIRCLE;
+			int player = -1;
 
-bool UInputMan::AccumulatedElementReleased(int element)
-{
-	if (element < 0 || element >= INPUT_COUNT)
-		return false;
+			// Retrieve deadzone settings from player's input scheme
+			for (int p = 0; p < UInputMan::MAX_PLAYERS; p++) {
+				int device = m_ControlScheme[p].GetDevice() - DEVICE_GAMEPAD_1;
+
+				if (device == joystick) {
+					deadZone = m_ControlScheme[p].GetJoystickDeadzone();
+					deadZoneType = m_ControlScheme[p].GetJoystickDeadzoneType();
+					player = p;
+					break;
+				}
+			}
+
+			// Disable input if it's in circle deadzone
+			if (player > -1 && deadZoneType == DeadZoneType::CIRCLE && deadZone > 0.0F) {
+				const InputMapping *element = m_ControlScheme[player].GetInputMappings();
+				Vector aimValues;
+
+				// Left stick
```

Comment:
Kill this comment. Also, for the love of god, clear out this repeated code. Either make a data structure (array/vector of pairs or whatever you want) and do a for loop over it, or split this out into another private method that takes the two input types, or do something else that works, cause goddamn what a waste of space!